src/fsspec_xrootd/xrootd.py

```diff
@@ -43,39 +61,184 @@ def _get_kwargs_from_urls(u: str) -> dict[Any, Any]:
+            if not status.ok:
+                raise OSError(f"File not touched properly: {status.message}")
+        else:
+            status, _ = self._myclient.truncate(
+                path, self.info(path).get("size"), timeout=self.timeout
+            )
+            if not status.ok:
+                raise OSError(f"File not touched properly: {status.message}")
+
+    def modified(self, path: str) -> Any:
+        status, statInfo = self._myclient.stat(path, timeout=self.timeout)
+        return statInfo.modtime
+
+    def sign(self, path: str, expiration: int = 100, **kwargs: Any) -> Any:
+        return (
+            self.storage_options["protocol"]
+            + "://"
+            + self.storage_options["hostid"]
+            + "/"
+            + self.storage_options["path_with_params"]
+        )
+
+    def exists(self, path: str, **kwargs: Any) -> bool | Any:
+        if path in self.dircache:
+            return True
+        else:
+            status, _ = self._myclient.stat(path, timeout=self.timeout)
+            if not status.ok:
+                if status.code == 400:
+                    return False
                 else:
-                    t = "file"
-
-                listing.append(
-                    {
-                        "name": path + "/" + item.name,
-                        "size": item.statinfo.size,
-                        "type": t,
-                    }
-                )
+                    raise OSError(f"status check failed with message: {status.message}")
+            else:
+                return True
+
+    def info(self, path: str, **kwargs: Any) -> dict[str, Any]:
+        spath = os.path.split(path)
+        deet = self._ls_from_cache(spath[0])
+        if deet is not None:
+            det = []
+            for item in deet:
+                if item["name"] == path:
+                    det.append(item)
+            return {
+                "name": path,
+                "size": det[0]["size"],
+                "type": det[0]["type"],
+            }
         else:
-            for item in deets:
-                listing.append(item.name)
+            status, deet = self._myclient.stat(path, timeout=self.timeout)
+            if not status.ok:
+                raise OSError(f"File stat request failed: {status.message}")
+            if deet.flags & StatInfoFlags.IS_DIR:
+                ret = {
+                    "name": path,
+                    "size": deet.size,
+                    "type": "directory",
+                }
+            elif deet.flags & StatInfoFlags.OTHER:
+                ret = {
+                    "name": path,
+                    "size": deet.size,
+                    "type": "other",
+                }
+            else:
+                ret = {
+                    "name": path,
+                    "size": deet.size,
+                    "type": "file",
+                }
+            _ = self.ls(spath[0], True, kwargs={"force_update": True})
```

Comment:
Is this to force the caching of the entire directory if it wasn't already? Since the `stat` call already gives the information for the specific path, doing this here may be unwanted by a client. Better to just take advantage of a cached `ls` call if it was previously made by the client than always call `ls`.

Reply:
ok